pkg/logql/syntax/lex.go

```diff
@@ -226,7 +228,7 @@ func (l *lexer) Lex(lval *exprSymType) int {
 		}
 	}
 
-	if tok, ok := functionTokens[tokenText]; ok {
+	if tok, ok := functionTokens[tokenTextLower]; ok {
```

Comment:
`isFunction` also needs a check for `by/without` tokens

Reply:
can you explain what you mean here exactly? they should be covered (see test case line 71)

Reply:
sorry I wasn't clear. I think `isFunction` would return false for `SUM BY (foo)` since we only check against lower case `by` and `without`
https://github.com/grafana/loki/blob/4c3e8f5958d8251bfba2ae2e807f1a75f7bdc01f/pkg/logql/syntax/lex.go#L396

Reply:
good catch! but `isFunction` is called only if `functionTokens[tokenTextLower]` is `ok`. But in case of by and without, it would never be called.. so now I'm wondering why is by and without a separate case in `isFunction`

Reply:
good catch!

Reply:
fixed this and added a test case to cover this. please verify.